### PR TITLE
user history

### PR DESCRIPTION
**Pull request by sabi1125**

### CODE REVIEW

File: app/src/controllers/client/user/UserController.ts [hunk 1]
```diff
@@ -59,7 +64,30 @@ const UserController: UserControllerInterface = {
...
+      reservations: reservations.map(r => ({
+        id: r.id,
+        shopId: r.shopId,
+        shopName: r.shopName,
+        menuName: r.menuName,
+        status: r.status,
+        reservationDate: convertDateTimeObjectToDateTimeString(r.reservationDate),
+        stylistName: r.StylistName,
+      })),
+    }
```
Comment (sabi1125) [inline app/src/controllers/client/user/UserController.ts:90]: could you plz look at this part once the user part i was trying to return with the reconstruct method but dont know why it gives an error saying cant find username and id basically all the user fields
Had to talk with my dad about school fees so took some time sorry abt that

File: app/src/controllers/client/user/UserController.ts [hunk 2]
```diff
@@ -5,16 +5,21 @@ import { Gender as EntityGender, User as EntityUser } from '@entities/User'
...
 export type UserServiceInterface = {
   fetchUserWithReservationCountAndReviewCount(id: number)
     : Promise<(EntityUser & { reservationCount: number, reviewCount: number })>
   signUpUser(email: string, username: string, password: string, confirm: string): Promise<EntityUser>
   updateUser(id: number, lastNameKanji: string, firstNameKanji: string, lastNameKana: string,
-    firstNameKana: string, gender: Gender, birthday: Date): Promise<EntityUser>
-  updateUserPassword(id: number, oldPassword: string, newPassword: string, confirmNewPassword: string)
-    : Promise<EntityUser>
+    firstNameKana: string, gender: Gender, birthday: Date): Promise<User>
+  updateUserPassword(id: number, oldPassword: string, newPassword: string, confirmNewPassword: string): Promise<User>
```
Comment (eugenesinamban) [inline app/src/controllers/client/user/UserController.ts:20]: you changed the return value of update user and update user password to User, which is a req-res-type model

File: app/src/controllers/client/user/UserController.ts (hunk 1 above)
Comment (eugenesinamban) [reply app/src/controllers/client/user/UserController.ts:91]:
```suggestion
    const reconstructedUser = await reconstructUser(user.id)
    const reservations = await UserService.fetchUserReservationsWithShopMenuAndStylistNames(user.id)
    return {
      user: reconstructedUser,
        id: u.id,
        username: u.username!,
        email: u.email,
        lastNameKana: u.lastNameKana,
        firstNameKana: u.firstNameKana,
        birthday: u.birthday ? convertDateObjectToOutboundDateString(u.birthday) : undefined,
        gender: u.gender ? convertEntityGenderToDTOGender(u.gender) : undefined,
        reservationCount: u.reservationCount,
        reviewCount: u.reviewCount,
      },
      reservations: reservations.map(r => ({
        id: r.id,
        shopId: r.shopId,
        shopName: r.shopName,
        menuName: r.menuName,
        status: r.status,
        reservationDate: convertDateTimeObjectToDateTimeString(r.reservationDate),
        stylistName: r.StylistName,
      })),
    }
```

File: app/src/controllers/client/user/repositories/ReservationRepository.ts [hunk 3]
```diff
@@ -1,10 +1,42 @@
...
 
 const ReservationRepository: ReservationRepositoryInterface = {
   async fetchUserReservationCount(userId) {
     return prisma.reservation.count({ where: { userId } })
   },
+
+  async fetchUserReservations(userId) {
+    const reservations = await prisma.reservation.findMany({
+      where: { userId },
+      take: 3,
```
Comment (eugenesinamban) [inline app/src/controllers/client/user/repositories/ReservationRepository.ts:27]: maybe we should separate the 3 logic somewhere else, not inside the repository

**Review by eugenesinamban — COMMENTED** [review]
LGTM

**Review by eugenesinamban — APPROVED** [review]

**Pull request merged**